compiler/rustc_builtin_macros/src/test.rs

```diff
@@ -252,6 +253,24 @@ pub fn expand_test_or_bench(
                                         "ignore",
                                         cx.expr_bool(sp, should_ignore(&cx.sess, &item)),
                                     ),
+                                    // ignore_message: Some("...") | None
+                                    field(
+                                        "ignore_message",
+                                        if let Some(msg) = should_ignore_message(cx, &item) {
+                                            if !cx.sess.features_untracked().ignore_message {
```

Comment:
This won't quite work -- this code doesn't run except during `--test`, which means that the gate here will be skipped over in normal compilation, effectively allowing the feature.

Testing on playground, it looks like we already allow (ungated) a reason to be passed for ignoring (https://github.com/rust-lang/rust/blob/9c3a3e3d5bd65fd534336101a06f7b11396c3208/compiler/rustc_feature/src/builtin_attrs.rs#L258) so we don't actually need to gate this feature -- it's just a matter of the message being passed into the test framework, which we can always just stop doing. Sorry for the runaround, I didn't check this earlier.

We can also drop the validation in should_ignore_message, I believe, because that's already checked by other code in the compiler.

Reply:
Good, thanks for helping me find out this, I did not see this either.  The things about the feature gate are removed.  Thanks for reviewing.